### ISSUE TITLE
stabil kréta-api

### ISSUE THREAD

**Issue by JeromeSchmied** [refilc/api-docs#2]
Azt meg tudnátok csinálni, hogy legyen egy használható, stabil api, ahol el lehet érni minden adatot a megfelelő `instituteCode`, felhasználónév, jelszó megadásával?

**Comment by kimaah** [comment]
ez mar adatkezelesnek minosulne a reszunkrol, amit egyelore nem tehetunk meg ceg nelkul, es utana is csak bonyolult modon, szoval nem hiszem hogy lesz ilyen, de ha lesz is nem mostanaban :(

**Comment by JeromeSchmied** [comment]
és akkor hogy tudnék csinálni egy `refilc-cli`-t Rustban?
mert átnéztem a dart(refilc/naplo), meg a ts(refilc/webnaplo) kódokat és bár nem nagyon értem, meg lehet csinálni az apit ugyanígy Rustban is ugye?

**Comment by ReinerRego** [comment]
Igen, itt megtalálsz mindent: https://github.com/bczsalba/ekreta-docs-v3
Esetleg ha szeretnéd, csinálunk neked egy repo-t a reFilc orgban.
Ha esetleg van még kérdésed, API-val kapcsolatban, mi hogyan mükszik, szólj

**Comment by WilliamNT** [comment]
> és akkor hogy tudnék csinálni egy `refilc-cli`-t Rustban? mert átnéztem a dart(refilc/naplo), meg a ts(refilc/webnaplo) kódokat és bár nem nagyon értem, meg lehet csinálni az apit ugyanígy Rustban is ugye?

Igen, lehetséges bármilyen nyelvben REST API klienset implementálni. A konkrét implementáció természetesen nyelv függő. Nézz utána, hogy Rust-ban milyen opciók állnak rendelkezésre kriptográfiára és webes kérésekre. Néhány kulcsszó, hogy könnyebb legyen utána nézz a dolgoknak: JWT, OAuth 2, REST, HTTP.

**Comment by JeromeSchmied** [comment]
> Igen, itt megtalálsz mindent: https://github.com/bczsalba/ekreta-docs-v3 Esetleg ha szeretnéd csinálunk neked egy repo-t a reFilc orgban. Ha van még kérdésed API-val kapcsolatban akár, szólj

Hát szerintem még nem állok neki rögtön, de majd pompás lenne egy repo a reFilcben.

Egyébként az nem baj, hogy a https://github.com/bczsalba/ekreta-docs-v3/blob/master/kreta_v2.py nem fut le?

**Comment by zypherift** [comment]
Alapból nem futna le magában.
Használd inkább importban. De egyszerűbb nézni szerintem a dokumentációt mint a python fájlt
Vagy cURL segítségével nézd meg mit ad vissza milyen endpoint.

**Comment by JeromeSchmied** [comment]
mostmár eljutottam odáig, hogy működik az `access_token` lekérés, így hát asszem készen áll a kód egy repora. Csináltok lécci? Valami refilc-rs, refilc-cli-rs vagy csak simán refilc-cli, amelyik jobban tetszik.

**Comment by kimaah** [comment]
tudsz csinalni sajat magadnak, majd ha teljesen kesz es jol mukodik, talan lesz rola egy fork az org alatt

**Comment by JeromeSchmied** [comment]
rendben

és ha pl [codeberg](https://codeberg.org)-en csinálok, akkor azt is át tudjátok rakni az org-ba?

**Comment by JeromeSchmied** [comment]
https://codeberg.org/jark/kreta-api-rs

**Comment by zypherift** [comment]
Tökéletes, biztos átlehet vinni

**Comment by JeromeSchmied** [comment]
> https://codeberg.org/jark/rsfilc

és csináltam hozzá [github mirror](https://github.com/JeromeSchmied/rsfilc)t

**Comment by B3ni15** [comment]
> meg még annyi, hogy mennyi `timeout`ot érdemes hagyni szegény kréta szervernek?

Általában nem kell neki semmi ahhoz képest gyorsan válaszol.
